go/oasis-node/cmd/debug/txsource/workload/runtime.go

```diff
@@ -0,0 +1,257 @@
+		return nil, fmt.Errorf("malformed tx output from runtime: %w", err)
+	}
+	if rsp.Error != nil {
+		return nil, fmt.Errorf("runtime tx failed: %s", *rsp.Error)
+	}
+
+	return &rsp, nil
+}
+
+func (r *runtime) doInsertRequest(ctx context.Context, rng *rand.Rand, rtc runtimeClient.RuntimeClient, existing bool) error {
+	key := r.generateVal(rng, existing)
+	value := r.generateVal(rng, false)
+
+	// Submit request.
+	rsp, err := r.submitRuntimeRquest(ctx, rtc, &runtimeTransaction.TxnCall{
+		Method: "insert",
+		Args: struct {
+			Key   string `json:"key"`
+			Value string `json:"value"`
+		}{
+			Key:   key,
+			Value: value,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("submit insert request failed: %w", err)
+	}
+	r.logger.Debug("insert request success",
+		"response", rsp,
+	)
+
+	if err := r.validateResponse(key, rsp); err != nil {
+		return fmt.Errorf("invalid response: %w", err)
+	}
+
+	// Update local state.
+	r.state[key] = value
+
+	return nil
+}
+
+func (r *runtime) doGetRequest(ctx context.Context, rng *rand.Rand, rtc runtimeClient.RuntimeClient, existing bool) error {
+	key := r.generateVal(rng, existing)
+
+	// Submit request.
+	rsp, err := r.submitRuntimeRquest(ctx, rtc, &runtimeTransaction.TxnCall{
+		Method: "get",
+		Args:   key,
+	})
+	if err != nil {
+		return fmt.Errorf("submit get request failed: %w", err)
+	}
+	r.logger.Debug("get request success",
+		"response", rsp,
+	)
+
+	if err := r.validateResponse(key, rsp); err != nil {
+		return fmt.Errorf("invalid response: %w", err)
+	}
+
+	return nil
+}
+
+func (r *runtime) doRemoveRequest(ctx context.Context, rng *rand.Rand, rtc runtimeClient.RuntimeClient, existing bool) error {
+	key := r.generateVal(rng, existing)
+
+	// Submit request.
+	rsp, err := r.submitRuntimeRquest(ctx, rtc, &runtimeTransaction.TxnCall{
+		Method: "remove",
+		Args:   key,
+	})
+	if err != nil {
+		return fmt.Errorf("submit remove request failed: %w", err)
+	}
+	r.logger.Debug("remove request success",
+		"response", rsp,
+	)
+
+	if err := r.validateResponse(key, rsp); err != nil {
+		return fmt.Errorf("invalid response: %w", err)
+	}
+
+	// Update local state.
+	delete(r.state, key)
+
+	return nil
+}
+
+func (r *runtime) Run(
+	gracefulExit context.Context,
+	rng *rand.Rand,
+	conn *grpc.ClientConn,
+	cnsc consensus.ClientBackend,
+	rtc runtimeClient.RuntimeClient,
+	fundingAccount signature.Signer,
+) error {
+	ctx := context.Background()
+
+	r.logger = logging.GetLogger("cmd/txsource/workload/runtime")
+	// Simple-keyvalue runtime.
+	err := r.runtimeID.UnmarshalHex(viper.GetString(CfgRuntimeID))
+	if err != nil {
+		r.logger.Error("runtime unmsrshal error",
+			"err", err,
+			"runtime_id", viper.GetString(CfgRuntimeID),
+		)
+		return fmt.Errorf("Runtime unmarshal: %w", err)
+	}
+	r.state = make(map[string]string)
+
+	for {
+		switch rng.Intn(3) {
+		case 0:
+			if err := r.doInsertRequest(ctx, rng, rtc, rng.Float64() < runtimeInsertExistingRatio); err != nil {
+				return err
```

Comment:
no context added, that it's an insert request

Reply:
fixed, also bellow